app/models/concerns/orchestration/compute.rb

```diff
@@ -207,14 +207,20 @@ def find_image
   end
 
   def validate_compute_provisioning
-    return true unless image_build?
-    return true if ( compute_attributes.nil? or (compute_attributes[:image_id] || compute_attributes[:image_ref]).blank? )
-    img = find_image
-    if img
-      self.image = img
+    return true if compute_attributes.nil?
+    if image_build?
```

Comment:
It's alright, I just thought this part of the if could be an external method and called from here like:

```ruby
if image_build?
  self.image = find_image 
else
  # delete image_id, image_ref from compute_attributes...
end
```

If it fails it's still fine. When you call `failure` etc... you're adding something to `.errors` so it'll fail. The method would also return false as `self.image` would be false.